[docs] Refactor "Selectors" (#4799)
DevExpress/testcafe: docs/articles/templates/selectors/selector-options.md

If the [visibilityCheck](#optionsvisibilitycheck) option is enabled, the element then must become visible within the `timeout`.

**Default value**: the timeout specified in the [runner.run](../../../using-testcafe/programming-interface/runner.md#run) API method or the [--selector-timeout](../../../using-testcafe/command-line-interface.md#--selector-timeout-ms) command line option.

### options.visibilityCheck

**Type**: Boolean

`true` to additionally require the returned element to become visible within [options.timeout](#optionstimeout).

This option is in effect when TestCafe waits for the selector to return a page element. This includes situations when

* a property is obtained from the selector;

    ```js
    const width = await Selector('#element', { visibilityCheck: true }).clientWidth;
    ```

* a selector property is passed to an [assertion](../../assertions/README.md) as its actual value;

    ```js
    await t.expect(Selector('#element', { visibilityCheck: true }).clientWidth).eql(400);
    ```

* a selector is evaluated with the `await` keyword;

    ```js
    const snapshot = await Selector('#element', { visibilityCheck: true })();
    ```

> If the target element is not visible, the selector throws an exception in all these cases.

Reviewer: Is `in all these cases.` needed here?  If we need to highlight that it related to only the situations listed above, maybe it's possible to rephrase this to make it a bit clearer?

Note that when a selector is passed to a [test action](../../actions/README.md) as the target element's identifier, the target element should be visible regardless of the `visibilityCheck` option.

Unlike filter functions, the `visibilityCheck` option does not change the selector's matched set.

Consider the following page:

```html
<html>
  <body>
    <div>This div is visible</div>
    <div style="display:none">This div not is visible</div>
    <div style="visibility:hidden">This div not is visible either</div>
  </body>
</html>
```

> When a selector with `visibilitycheck` enabled is tested for an element existance or the number of matching elements, invisible elements also count.

Reviewer: `existance` has a type, should be `existence`

## Options

### options.boundTestRun

**Type**: Object

If you need to call a selector from a Node.js callback, assign the current [test controller](../../test-code-structure.md#test-controller) to the `boundTestRun` option.

> For details, see [Calling Selectors from Node.js Callbacks](edge-cases-and-limitations.md#calling-selectors-from-nodejs-callbacks).

Reviewer 2: ```suggestion
For more information, see [Calling Selectors from Node.js Callbacks](edge-cases-and-limitations.md#calling-selectors-from-nodejs-callbacks).
```

});
```

### options.timeout

**Type**: Number

> The amount of time, in milliseconds, allowed for an element returned by the selector to appear in the DOM before the test fails.

Reviewer 2: ```suggestion
The time (in milliseconds) allowed for an element returned by the selector to appear in the DOM before the test fails.
```

</html>
```

> When a selector with `visibilitycheck` enabled is tested for an element existence or the number of matching elements, invisible elements also count.

Reviewer 2: "an element existence"?

Reviewer 2: invisible elements also count -> invisible elements are also counted

Author: *When you use a selector with `visibilitycheck` enabled to determine if an element exists or to count matching elements, TestCafe also takes into account invisible elements.*

Reviewer 2: Ok

// does not affect the selector's matched set
```

> In case you need to filter page elements by their visibility, use [filterVisible](functional-style-selectors.md#filtervisible) and [filterHidden](functional-style-selectors.md#filterhidden) methods.

Reviewer 2: ```suggestion
To filter page elements by their visibility, use [filterVisible](functional-style-selectors.md#filtervisible) and [filterHidden](functional-style-selectors.md#filterhidden) methods.
```
